work around a weird shell quirk, it doesn't fall in the X) case if
with_cpu is unset/empty, but in the *) case instead, causing an abort,
bug #416577

--- gcc-4.2.4/gcc/config.gcc
+++ gcc-4.2.4/gcc/config.gcc
@@ -1216,7 +1216,7 @@
 		need_64bit_hwint=yes
 		# FIXME: -m64 for i[34567]86-*-* should be allowed just
 		# like -m32 for x86_64-*-*.
-		case X"${with_cpu}" in
+		case X"${with_cpu:-generic}" in
 		Xgeneric|Xnocona|Xx86-64|Xk8|Xopteron|Xathlon64|Xathlon-fx)
 			;;
 		X)
